lib/boot.ts

```diff
@@ -1,246 +1,81 @@
-  version: 42,
-});
-
-const debug = Debug('client');
-const l = (msg: string) => (...args: unknown[]) => debug(msg, ...args);
-
-client
-  .on('connect', l('Connected'))
-  .on('disconnect', l('Not connected'))
-  .on('message', l('<='))
-  .on('send', l('=>'))
-  .on('unauthorized', l('Not authorized'));
 
-client.on('unauthorized', () => {
-  // If a token exists, we probaly reached this point from a password change.
-  // The client should sign out the user, but preserve db content in case
-  // some data has not synced yet.
-  if (token) {
-    client.clearAuthorization();
-    redirectToWebSigninIfNecessary();
-    return;
+  const storedUser = JSON.parse(storedUserData);
+  if (storedUser?.accessToken && storedUser?.username) {
+    return [storedUser.accessToken, storedUser.username];
   }
 
-  client.reset().then(() => {
-    console.log('Reset complete'); // eslint-disable-line no-console
-  });
-});
-
-let props = {
-  client: client,
-  noteBucket: client.bucket('note'),
-  preferencesBucket: client.bucket('preferences'),
-  tagBucket: client.bucket('tag'),
-  isDevConfig: isDevConfig(config?.development),
-  onAuthenticate: (username: string, password: string) => {
-    if (!(username && password)) {
-      return;
-    }
+  return [null, null];
+};
 
-    store.dispatch(setPendingAuth());
-    auth
-      .authorize(username, password)
-      .then((user) => {
-        resetStorageIfAccountChanged(username);
-        if (!user.access_token) {
-          return store.dispatch(resetAuth());
-        }
+const [storedToken, storedUsername] = getStoredAccount();
 
-        store.dispatch(setAccountName(username));
-        store.dispatch(setAuthorized());
-        localStorage.access_token = user.access_token;
-        token = user.access_token;
-        client.setUser(user);
-        analytics.tracks.recordEvent('user_signed_in');
-      })
-      .catch(({ message }: { message: string }) => {
-        if (
-          some([
-            'invalid password' === message,
-            message.startsWith('unknown username:'),
-          ])
-        ) {
-          store.dispatch(setInvalidCredentials());
-        } else {
-          store.dispatch(setLoginError());
-        }
-      });
-  },
-  onCreateUser: (username: string, password: string) => {
-    if (!(username && password)) {
-      return;
-    }
+if (config.is_app_engine && !storedToken) {
+  window.webConfig?.signout?.(() => {
+    window.location = `${config.app_engine_url}/`;
+  });
+}
 
-    store.dispatch(setPendingAuth());
-    auth
-      .create(username, password, appProvider)
-      .then((user) => {
-        resetStorageIfAccountChanged(username);
-        if (!user.access_token) {
-          return store.dispatch(resetAuth());
-        }
+const run = (
+  token: string | null,
+  username: string | null,
+  createWelcomeNote: boolean
+) => {
+  if (token) {
+    import('./boot-with-auth').then(({ bootWithToken }) => {
+      bootWithToken(
+        () => {
```

Comment:
boot feels like the wrong place to have all this logic. Not now but in future PR's how do we move all this out of boot.

Reply:
What feels wrong about it?

Reply:
The code to sign a user out exists in the boot file. Analytics should be moved to middleware. Logging out functionality should be moved to an action.

Reply:
will discuss in Slack, but for the sake of responding here…

> the code to sign a user out exists in the boot file

yeah I did this intentionally, specifically with the `cookie` handling. I wanted to move all the persistence and clearing of the authentication into one place and at the top so we can control how to load the app (with a side benefit of slimming up the logged-out page load)

if we move it out of `boot` where would you want it to move?

> Analytics should be moved to middleware

this is something I discovered while doing this work - we're sending out analytics before we boot the app, even in `develop`. the `analytics` module appears to queue up these events until we load and until we have a reading on `analytics_enabled`, which is good, but it definitely _looks_ wrong to be calling for trackers before being logged in (it looks like we're not waiting for consent even though we are)

if we want these edge calls to be in middleware then we're going to need to initialize the Redux store before login (which is what currently happens in `develop`). If we do that, I'd still really like to replace the store during the login, which we can do with `replaceReducer()`

I actually thought about doing this when creating this PR but I wanted to avoid the complexity of creating that replacement mechanism. if we move this in a future PR, I guess that would be a reasonable way to go about it.

> Logging out functionality should be moved to an action.

It currently is handled by an action within the app but the Simperium middleware calls this function in response to that action or in response to an authentication expiration.